controllers/fenceagentsremediation_controller_test.go

```diff
@@ -0,0 +1,214 @@
+})
+
+
+// buildFARTemplate from string to arrays to two string maps (key-value manner)
+func buildFARTemplate(fields []string, values []string, nodeFields []string, nodeValues []string, node string) (map[v1alpha1.ParameterName]string, map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) {
+	testShareParam := make(map[v1alpha1.ParameterName]string)
+	testNodeParam := make(map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string)
+	i := 0
+	for i = 0; i < len(values); i++ {
+		field := v1alpha1.ParameterName(fields[i])
+		testShareParam[field] = values[i]
+	}
+
+	nodeName := v1alpha1.NodeName(node)
+	numNodeParam := len(fields) - len(values)
+	for j := 0; j < numNodeParam; j++ {
+		if indexOf(node, nodeFields) > -1 {
+			field := v1alpha1.ParameterName(fields[i+j])
+			testNodeParam[field] = make(map[v1alpha1.NodeName]string)
+			testNodeParam[field][nodeName] = nodeValues[indexOf(node, nodeFields)]
+		}
+	}
+	return testShareParam, testNodeParam
+}
+
+// indexOf return the index of element in data array. If it is not found, return -1
+func indexOf(element string, data []string) int {
+	for k, v := range data {
+		if element == v {
+			return k
+		}
+	}
+	return -1 
+}
+
+// newFenceAgentsRemediationTemplate assign the input to the FenceAgentsRemediationTemplate's Spec
+func newFenceAgentsRemediationTemplate(agent string, sharedparameters map[v1alpha1.ParameterName]string, nodeparameters map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) *v1alpha1.FenceAgentsRemediationTemplate {
+	return &v1alpha1.FenceAgentsRemediationTemplate{
+		ObjectMeta: metav1.ObjectMeta{Name: fenceAgentsTemplateName, Namespace: defaultNamespace},
+		Spec: v1alpha1.FenceAgentsRemediationTemplateSpec{
+			Agent:            agent,
+			SharedParameters: sharedparameters,
+			NodeParameters:   nodeparameters,
+		},
+	}
+}
+
+// buildFarPod builds a dummy pod with FAR label and namespace
+func buildFarPod() * corev1.Pod {
+	fenceAgentsPod := &corev1.Pod{}
+	fenceAgentsPod.Labels = faPodLabels
+	fenceAgentsPod.Name = "mock-fence-agents"
+	fenceAgentsPod.Namespace = defaultNamespace
+	container := corev1.Container{
+		Name:  "foo",
+		Image: "foo",
+	}
+	fenceAgentsPod.Spec.Containers = []corev1.Container{container}
+	return fenceAgentsPod
+}
+
+// cliCommandsEquality creates the command for CLI and compares it with the production command
+func cliCommandsEquality(farTemplate *v1alpha1.FenceAgentsRemediationTemplate, far *v1alpha1.FenceAgentsRemediation) bool {
+	//fence_ipmilan --ip=192.168.111.1 --ipport=6233 --username=admin --password=password --action=status --lanplus
+	command, err := buildFenceAgentParams(farTemplate, far)
+	Expect(err).NotTo(HaveOccurred())
+	command = append([]string{farTemplate.Spec.Agent}, command...)
+	Expect(executedCommand).ToNot(Equal(nil))
+	fmt.Printf("%s is the executedCommand in prod, and %s is the expected command in test.\n", executedCommand, command)
+	return contains(executedCommand, command) && contains(command, executedCommand)
```

Comment:
I think This method is incorrect, check playground example [here](https://go.dev/play/p/VdqRfOXwPbg).
I suggest replacing `contains(executedCommand, command) && contains(command, executedCommand)` with a single method: (feel free to use better naming)
```
func isContainSameElements(s1 []string, s2 []string) bool {
	if len(s1) != len(s2) {
		return false
	}
	s1ElementsAndCount, s2ElementsAndCount := make(map[string]int), make(map[string]int)
	for i := 0; i < len(s1); i++ {
		s1ElementsAndCount[s1[i]]++
		s2ElementsAndCount[s2[i]]++
	}

	for key, _ := range s1ElementsAndCount {
		if s1ElementsAndCount[key] != s2ElementsAndCount[key] {
			return false
		}
	}
	return true

}
```

Reply:
Good example, and better function with much lower complexity (o(n) vs o(n^2))

Reply:
```
sort.Strings(s1)
sort.Strings(s2)
return reflect.DeepEqual(s1, s2)
```